src/ng/rootScope.js

```diff
@@ -353,11 +353,13 @@ function $RootScopeProvider() {
        *    - `newVal` contains the current value of the `watchExpression`
        *    - `oldVal` contains the previous value of the `watchExpression`
        *    - `scope` refers to the current scope
-       * @param {boolean=} objectEquality Compare for object equality using {@link angular.equals} instead of
-       *     comparing for reference equality.
+       * @param {boolean=|function()} objectEquality Compare for object equality using {@link angular.equals} instead of
+       *     comparing for reference equality. If a function is passed, it will be used as a replacement for {@link angular.equals}.
+       * @param {function()} copier Replacement function for {@link angular.copy} to save value into oldVal for next
+            change event.
        * @returns {function()} Returns a deregistration function for this listener.
        */
-      $watch: function(watchExp, listener, objectEquality) {
+      $watch: function(watchExp, listener, objectEquality, copier) {
```

Comment:
To make it clear that if you are going to mess with the way equality works you also have to understand that copy will likely need to change too